src/lsptoolshost/server/roslynLanguageServer.ts

```diff
@@ -617,15 +617,23 @@ export class RoslynLanguageServer {
                 ? path.join(context.extension.extensionPath, '.razor')
                 : razorOptions.razorServerPath;
 
-        args.push('--razorSourceGenerator', path.join(razorPath, 'Microsoft.CodeAnalysis.Razor.Compiler.dll'));
+        let razorComponentPath = '';
+        getComponentPaths('razorExtension', languageServerOptions).forEach((extPath) => {
+            additionalExtensionPaths.push(extPath);
+            razorComponentPath = path.dirname(extPath);
+        });
+
+        // If cohosting is enabled we get the source generator from the razor component path
```

Comment:
Shouldn't they be the same?

Reply:
In a real build, yes they'd be the same, but for local development not necessarily. If you're only working on cohosting, then there is no need to specify a local path for rzls at all, since it won't run. Plus this makes more sense in future when the .razor folder will presumably go away entirely.